source/handshake.c

```diff
@@ -401,6 +407,25 @@ int main(int argc, char * const argv[])
             l_flag = 1;
             test_case = TC_SSL_CTX_POOL;
             break;
+        case 'S': {
+            char *end = NULL;
+            errno = 0;
+            int sec_mem_size = (int)strtol(optarg, &end, 10);
+            if (errno || !end || *end || sec_mem_size <= 0) {
```

Comment:
Nit: Please add empty line after declarations and use `end == NULL` instead of `!end`